### ISSUE TITLE
Unexpected error when passing a type to an unidentified preprocessor definition

### ISSUE THREAD

**Issue by Anthony-Nicholls** [llvm/llvm-project#140057]
Given the following code...

```cpp
class T{};

void f()
{
    UNDEFINED_MACRO (T);
}
```
https://godbolt.org/z/5WjGM4KE6

we might expect an error regarding the undeclared identifier `UNDEFINED_MACRO` however, in this specific case we get `'T' does not refer to a value`. 

If we change the code to so that it passes a variable instead of a type, for example...

```cpp
void f()
{
    int i;
    UNDEFINED_MACRO (i);
}
```
https://godbolt.org/z/YT1GGPG4Y

then the error is the more expected `use of undeclared identifier 'UNDEFINED_MACRO'`.

gcc and MSVC both appear to error as we might expect in this scenario.

I've tested this using trunk on CompilerExplorer as linked above.

**Comment by shafik** [comment]
Diagnostic is `err_ref_non_value`

Note, we seem to do a better job in C: https://godbolt.org/z/d389Y5a9r
